### PR TITLE
Cleaned contract and testing, updated readme, decouple verify

### PR DESCRIPTION
**Pull request by Raihan9797**
Remove unecessary commented out code from contract and test file
Update readme to install and run our code
decouple verify() into a check_user_list(), and verify().

### CODE REVIEW

File: test/test_habit.js [hunk 1]
```diff
@@ -135,55 +140,34 @@ contract('Habit', function(accounts) {
...
         // console.log(user1_day4_verified);
-        // let is_user1_array2 = await habit_instance.get_user_check_list(0, accounts[1]);
         /* check user1 array after 5 days and skipping day 4
-        let is_user1_array2 = await habit_instance.get_user_check_list.call(0, accounts[1]);
+        */
+        let is_user1_array = await habit_instance.get_user_check_list.call(0, accounts[1]);
         // loop through array of big numbers
-        for (let i = 0; i < is_user1_array2.length; i++) {
-            console.log(is_user1_array2[i].toNumber());
+        console.log('user 1 checklist')
```
Comment (ambhinav) [inline test/test_habit.js:168]: change to event check

File: test/test_habit.js [hunk 2]
```diff
@@ -135,55 +140,34 @@ contract('Habit', function(accounts) {
...
-            is_user1_loser2,
-            true,
-            "users check_list not filled properly 2"
-        )
-    });
+        // only tick day2 for user 2
+        let user2_day2_verified = await habit_instance.tick_user_list(0, accounts[2], 2);
 
+        let is_user2_array = await habit_instance.get_user_check_list.call(0, accounts[2]);
+        console.log('user 2 checklist')
```
Comment (ambhinav) [inline test/test_habit.js]: change to event check

**Review by ambhinav — CHANGES_REQUESTED** [review]

**Comment by Raihan9797** [comment]
Replace console.log with emit events in testing
Implemented transfer to main pool when all users lose a habit challenge
Contract owner can withdraw() from the main_pool
Implemented testing for these questions

**Review by dragontho — APPROVED** [review]

**Review by ambhinav — APPROVED** [review]
LGTM

**Pull request merged**